doc/api/wasi.md

```diff
@@ -0,0 +1,85 @@
+# WebAssembly System Interface (WASI)
+
+<!--introduced_in=REPLACEME-->
+
+> Stability: 1 - Experimental
+
+The WASI API provides an implementation of the [WebAssembly System Interface][]
```

Comment:
I think an explanation that WASI is basically a thing that lets us use file and socket I/O from WASM in Node.js might be useful here.

Most of our users probably aren't even sure what POSIX or system calls are.

Reply:
@benjamingr any chance you could provide the specific text you'd like to see? You can leave it as a comment or push directly to the branch as you see fit.

Reply:
ping @benjamingr